packages/frontend/components/scaffold-eth/Contract/ContractUI.tsx

```diff
@@ -0,0 +1,69 @@
+import { Contract } from "ethers";
+import { useContract, useNetwork, useProvider } from "wagmi";
+import {
+  getAllContractFunctions,
+  getContractReadOnlyMethodsWithParams,
+  getContractVariablesAndNoParamsReadMethods,
+  getContractWriteMethods,
+  getDeployedContract,
+} from "./utilsContract";
+
+type ContractProps = {
+  contractName: string;
+};
+
+/**
+ * TODO
+ * handle error
+ * handle types correctly
+ * handle loading state
+ * handle payable, public, functions
+ **/
+const ContractUI = ({ contractName }: ContractProps) => {
+  const { chain } = useNetwork();
+  const provider = useProvider();
+
+  let contractAddress = "";
+  let contractABI = [];
+  const deployedContractData = getDeployedContract(chain?.id.toString(), contractName);
+
+  if (deployedContractData) {
+    ({ address: contractAddress, abi: contractABI } = deployedContractData);
+  }
+
+  const contract: Contract = useContract({
+    addressOrName: contractAddress,
+    contractInterface: contractABI,
+    signerOrProvider: provider,
+  });
+
+  const displayedContractFunctions = getAllContractFunctions(contract);
+
+  const contractVariablesDisplay = getContractVariablesAndNoParamsReadMethods(contract, displayedContractFunctions);
+  const contractMethodsDisplay = getContractReadOnlyMethodsWithParams(contract, displayedContractFunctions);
+  const contractWriteMethods = getContractWriteMethods(contract, displayedContractFunctions);
+
+  if (!contractAddress) {
+    return <p className="text-2xl">No Contract found !</p>;
```

Comment:
I removed the `text-white` here because I wasn't seeing it while testing. I think you are using "dark mode" right?

Dark / light mode it's definitely something we have to implement on tailwind / daisy. We can create an issue for it!

Reply:
Yup using dark mode, currently it's dark/white mode is based on the system theme